frontend/public/components/secrets/create-secret.tsx

```diff
@@ -0,0 +1,155 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import { k8sCreate, k8sUpdate, K8sResourceKind } from '../../module/k8s';
+import { ButtonBar, Firehose, history, kindObj, StatusBox } from '../utils';
+import { getActiveNamespace, formatNamespacedRouteForResource, UIActions } from '../../ui/ui-actions';
+import { SafetyFirst } from '../safety-first';
+import { WebHookSecretSubform } from './subforms';
+
+const determineSecretTypeAbstraction = (secret) => {
```

Comment:
Might be fine to name this `determineSecretType()` or just `secretType()`.

Reply:
Secrets are a kind already have `type` field, and all the values that could return from that function will be just an type abstractions:
- webhook
- source
- image
- key/value

Reply:
Ok makes sense.